admin/links/class-link-watcher.php

```diff
@@ -21,37 +21,77 @@ public function __construct( WPSEO_Link_Content_Processor $content_processor ) {
 	}
 
 	/**
-	 * Registers the hooks
+	 * Registers the hooks.
+	 *
+	 * @returns void
 	 */
 	public function register_hooks() {
-		add_action( 'save_post', array( $this, 'save_post' ), 10, 2 );
+		add_action( 'transition_post_status', array( $this, 'post_status_transition' ), 10, 3 );
```

Comment:
Why did we remove `save_post` here?
We still need to count the links when the post is being updated.